src/codeflare_sdk/common/kubernetes_cluster/kube_api_helpers.py

```diff
@@ -19,34 +19,45 @@
 
 import executing
 from kubernetes import client, config
-from urllib3.util import parse_url
-import os
 
 
 # private methods
 def _kube_api_error_handling(
     e: Exception, print_error: bool = True
 ):  # pragma: no cover
-    perm_msg = (
-        "Action not permitted, have you put in correct/up-to-date auth credentials?"
-    )
-    nf_msg = "No instances found, nothing to be done."
-    exists_msg = "Resource with this name already exists."
-    if type(e) == config.ConfigException:
-        raise PermissionError(perm_msg)
-    if type(e) == executing.executing.NotOneValueFound:
+    def print_message(message: str):
         if print_error:
-            print(nf_msg)
-        return
-    if type(e) == client.ApiException:
-        if e.reason == "Not Found":
-            if print_error:
-                print(nf_msg)
-            return
-        elif e.reason == "Unauthorized" or e.reason == "Forbidden":
-            if print_error:
-                print(perm_msg)
-            return
-        elif e.reason == "Conflict":
-            raise FileExistsError(exists_msg)
-    raise e
+            print(message)
+
+    error_messages = {
+        "Not Found": "The requested resource could not be located.\n"
+        "Please verify the resource name and namespace.",
+        "Unauthorized": "Access to the API is unauthorized.\n"
+        "Check your credentials or permissions.",
+        "Forbidden": "Access denied to the Kubernetes resource.\n"
+        "Ensure your role has sufficient permissions for this operation.",
+        "Conflict": "A conflict occurred with the RayCluster resource.\n"
+        "Only one RayCluster with the same name is allowed."
+        "Please delete or rename the existing RayCluster before creating a new one with the desired name.",
+    }
```

Comment:
Small nit, can we move this to be a constant outside the function otherwise I tried this out and it LGTM 👍🏻